src/classes/MyHandler/MyHandler.ts

```diff
@@ -678,6 +680,12 @@ export default class MyHandler extends Handler {
                     } else if (sku === '5021;6') {
                         exchange.contains.keys = true;
                         exchange[which].contains.keys = true;
+                        if (which === 'our') {
+                            keyOurSide = true;
+                        } else if (which === 'their' && keyOurSide === true) {
```

Comment:
Are you sure that `else if` should be here and not just `if`?

Reply:
How can I determine that there are keys on both sides if I didn't do like that?

Reply:
Right, now I've managed to catch the logic. Changing `else if` to `if` wouldn't change anything, since `which` can't be `their` and `our` at the same time. But I still kinda dislike the logic. Because if for some reason we change the order of processing sides (so that "their" is processed before "our"), this feature will break.

Reply:
Yea, anyway gonna keep it for now.